File: src/main/java/com/eternalcode/plots/member/MemberRepository.java
package com.eternalcode.plots.member;

import java.util.UUID;
import java.util.concurrent.CompletableFuture;

public interface MemberRepository {

    CompletableFuture<Void> saveMember(Member member);

    CompletableFuture<Void> removeMember(Member member);

    CompletableFuture<Boolean> isMember(UUID plotId, UUID userId);

}

File: src/main/java/com/eternalcode/plots/member/MemberRepository.kt
package com.eternalcode.plots.member

import java.util.*
import java.util.concurrent.CompletableFuture

interface MemberRepository {
    fun saveMember(member: Member): CompletableFuture<Void?>?

    fun removeMember(member: Member): CompletableFuture<Void?>?

    fun isMember(plotId: UUID, userId: UUID): CompletableFuture<Boolean>?
}
